File: messaging/grpc/explainer/ruby/explain.rb
#!/usr/bin/env ruby

this_dir = File.expand_path(File.dirname(__FILE__))
lib_dir = File.join(this_dir, 'lib')
$LOAD_PATH.unshift(lib_dir) unless $LOAD_PATH.include?(lib_dir)

require 'grpc'
require 'explainer_services'

def main
  stub = Explainer::ShiFu::Stub.new('localhost:50051', :this_channel_is_insecure)
  problem = ARGV.size > 0 ?  ARGV[0] : 'help'
  puts stub.tell_me_why(Explainer::ExplainerRequest.new(problem: problem)).explanation
end

main

File: messaging/grpc/explainer/python/explain.py
from __future__ import print_function
from sys import argv
from grpc.beta import implementations

import explainer_pb2

_TIMEOUT_SECONDS = 10


def run(problem):
  channel = implementations.insecure_channel('localhost', 50051)
  stub = explainer_pb2.beta_create_ShiFu_stub(channel)
  response = stub.TellMeWhy(explainer_pb2.ExplainerRequest(problem=problem), _TIMEOUT_SECONDS)
  print(response.explanation)


if __name__ == '__main__':
  run(argv[1] if len(argv) > 1 else 'help')
